packages/lib/payment/handlePaymentSuccess.ts

```diff
@@ -125,10 +29,16 @@ export async function handlePaymentSuccess(paymentId: number, bookingId: number)
     bookingData.references = { create: scheduleResult.referencesToCreate };
   }
 
-  if (eventTypeRaw?.requiresConfirmation) {
+  const requiresConfirmation = doesBookingRequireConfirmation({
```

Comment:
Correctly determines requiresConfirmation for booking consider there can be threshold case as well.